homeassistant/components/boschshc/__init__.py

```diff
@@ -0,0 +1,136 @@
+"""The Bosch Smart Home Controller integration."""
+import asyncio
+import logging
+
+from boschshcpy import SHCSession
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    CONF_IP_ADDRESS,
+    CONF_NAME,
+    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+
+from .const import CONF_SSL_CERTIFICATE, CONF_SSL_KEY, DOMAIN
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_NAME, default="Bosch SHC"): cv.string,
+                vol.Required(CONF_IP_ADDRESS): cv.string,
+                vol.Required(CONF_SSL_CERTIFICATE): cv.isfile,
+                vol.Required(CONF_SSL_KEY): cv.isfile,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+PLATFORMS = [
+    "binary_sensor",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Bosch SHC component."""
+    hass.data.setdefault(DOMAIN, {})
+    conf = config.get(DOMAIN)
+
+    if not conf:
+        return True
+
+    configured_hosts = {
+        entry.data.get(CONF_IP_ADDRESS)
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+    if conf[CONF_IP_ADDRESS] in configured_hosts:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+        )
+    )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Bosch SHC from a config entry."""
+    data = entry.data
+
+    session = await hass.async_add_executor_job(
+        SHCSession,
+        data[CONF_IP_ADDRESS],
+        data[CONF_SSL_CERTIFICATE],
+        data[CONF_SSL_KEY],
+    )
+
+    shc_info = session.information
+    if shc_info.version == "n/a":
+        _LOGGER.warning("Unable to connect to Bosch Smart Home Controller API")
+        return False
```

Comment:
Is this a non recoverable error or just a temporary connection problem? If it's the latter we could raise `ConfigEntryNotReady`.